dns/nameserver.go

```diff
@@ -0,0 +1,85 @@
+package dns
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+)
+
+// Nameserver represents a DNS nameserver.
+type Nameserver struct {
+	// IPAddr is the IP address of the nameserver.
+	IP net.IP
+
+	// Port is the port of the nameserver.
+	Port uint16
+}
+
+// Addr returns the address of the nameserver as a string.
+func (n Nameserver) Addr() string {
+	return n.IP.String() + ":" + strconv.Itoa(int(n.Port))
+}
+
+// ParseNameserverAddr parses a nameserver address string into an IP and a port.
+//
+// It expects the `addr` to be in the format `ip` or `ip[:port]`. Where `ip` can be an IPv4 or an IPv6 address.
+func ParseNameserverAddr(addr string) (Nameserver, error) {
```

Comment:
```suggestion
func parseNameserverAddr(addr string) (Nameserver, error) {
```
Is it needed as exported?